packages/framework/src/Model/Article/ArticleFacade.php

```diff
@@ -109,6 +109,64 @@ public function getOrderedArticlesByDomainIdAndPlacementQueryBuilder($domainId,
         return $this->articleRepository->getOrderedArticlesByDomainIdAndPlacementQueryBuilder($domainId, $placement);
     }
 
+    /**
+     * @param int $domainId
+     * @return int
+     */
+    public function getAllVisibleArticlesCountByDomainId(int $domainId): int
+    {
+        return $this->articleRepository->getAllVisibleArticlesCountByDomainId($domainId);
+    }
+
+    /**
+     * @param int $domainId
+     * @param string $placement
+     * @return int
+     */
+    public function getAllVisibleArticlesCountByDomainIdAndPlacement(int $domainId, string $placement): int
+    {
+        return $this->articleRepository->getAllVisibleArticlesCountByDomainIdAndPlacement($domainId, $placement);
+    }
+
+    /**
+     * @param int $domainId
+     * @param int $limit
+     * @param int $offset
+     * @return \Shopsys\FrameworkBundle\Model\Article\Article[]
+     */
+    public function getVisibleArticlesSortedByPlacementAndPositionByDomainIdLimited(
+        int $domainId,
+        int $limit,
+        int $offset
+    ): array {
+        return $this->articleRepository->getVisibleSortedByPlacementAndPositionByDomainIdLimited(
+            $domainId,
+            $limit,
+            $offset
+        );
+    }
+
+    /**
+     * @param int $domainId
+     * @param string $placement
+     * @param int $limit
+     * @param int $offset
+     * @return \Shopsys\FrameworkBundle\Model\Article\Article[]
+     */
+    public function getVisibleArticlesForPlacementByDomainIdLimited(
```

Comment:
```suggestion
    public function getVisibleArticlesForPlacement(
```
or
```suggestion
    public function getVisibleArticlesListForPlacement(
```

Reply:
Names of repository methods should be updated too

Reply:
Again.

I do not want to drop the information about domain specification and the fact is is returning some part of list.

Maybe `getVisibleArticlesListForPlacementByDomainId` or `getVisibleArticlesListByPlacementAndDomainId`?

Reply:
```suggestion
    public function getVisibleArticlesByDomainIdAndPlacementLimited(
```

Reply:
fixed